packer/conf/buildkite-agent/scripts/fix-buildkite-agent-builds-permissions

```diff
@@ -1,81 +1,59 @@
 #!/bin/bash
-
-# Files that are created by Docker containers end up with strange user and
-# group ids, usually 0 (root). Docker namespacing will one day save us, but it
-# can only map a single docker user id to a given user id (not any docker user
-# id to a single system user id).
-#
-# Until we can map any old user id back to
-# buildkite-agent automatically with Docker, then we just need to fix the
-# permissions manually before each build runs so git clean can work as
-# expected.
-
 set -eu -o pipefail
 
-# We need to scope this to only the currently running agent's builds, but we
-# also need to control security and make sure arbitrary folders can't be
-# chmoded.
-#
-# We prepare the agent build directory basename in the environment hook and
-# pass it as the first argument. In here we just need to check that it contains
-# no slashes and isn't a traversal component.
-AGENT_BUILDS_NAME="$1"
-# => "my-agent-1"
+### Overview
 
-# Make sure it doesn't contain any slashes by substituting slashes with nothing
-# and making sure it doesn't change
-if [[ "${AGENT_BUILDS_NAME//\//}" != "${AGENT_BUILDS_NAME}" ]]; then
-	exit 1
-fi
+# When docker is used in builds, the files created in containers that make it back to the
+# build directories by way of volume mounts have the UID and GID of the docker daemon. This
+# is typically root(0), which results in permissions problems with the `git clean` step
+# of builds (as the buildkite agent runs as buildkite-agent).
 
-# Now we know this name has only non-slash characters.
-#
-# We just need to check that it's not "." or ".." to prevent traversal:
-if [[ "${AGENT_BUILDS_NAME}" == "." || "${AGENT_BUILDS_NAME}" == ".." ]]; then
-	exit 2
-fi
+# The long term solution to this problem is to use username space remapping in docker, but
+# this is not always available and there are some quirks around how containers with multiple
+# users inside them are mapped to host users.
 
-# We also need to make sure it's not empty so we don't collide with other agents:
-if [[ -z "${AGENT_BUILDS_NAME}" ]]; then
-	exit 3
-fi
+# This script can be run in a Buildkite environment hook to reset permissions for an agents
+# builds directory to avoid errors.
 
-# We know the builds path:
-BUILDS_PATH="/var/lib/buildkite-agent/builds"
```

Comment:
The reason to put this in here was to absolutely make sure that there was no way for this script to be used anywhere but inside this known part of the filesystem.